src/pages/Auth/Login.js

```diff
@@ -0,0 +1,139 @@
+import axios from "axios";
+import React, { useState } from "react";
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useAuth } from "../../contexts";
+import { useDocumentTitle, useToast } from "../../hooks";
+import "./Auth.css";
+
+export function Login() {
+  const [error, setError] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [loginForm, setLoginForm] = useState({
+    email: "",
+    password: "",
+  });
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { dispatch } = useAuth();
+  const { showToast } = useToast();
+  useDocumentTitle("Login");
+
+  const handleFormHandler = async (event) => {
+    event.preventDefault();
+    try {
+      const response = await axios.post("/api/auth/login", loginForm);
+      const { foundUser: user, encodedToken } = response.data;
+      dispatch({ type: "AUTH_SUCCESS", payload: { user, encodedToken } });
+      localStorage.setItem("user", JSON.stringify(user));
+      localStorage.setItem("token", encodedToken);
+      if (location.state !== null) {
+        navigate(location?.state?.from?.pathname);
+      } else {
+        navigate("/");
+      }
+      showToast("Logged In!", "success");
+    } catch (error) {
+      showToast(error.response.data.errors[0], "error");
+      setError("Email or password is incorrect");
+    }
+  };
+
+  function loginHandler() {
+    setLoginForm({ email: "ranitasaha21@gmail.com", password: "ranitasaha" });
+  }
+  return (
+    <div className="section__login">
+      <div className="form-box">
+        <div className="button-box">
+          <div id="btn"></div>
+          <h3 className="toggle-btn">LOGIN</h3>
+        </div>
+        <form className="input-group login" onSubmit={handleFormHandler}>
+          <label className="password-label" htmlFor="mail">
+            Email Address
+          </label>
+          <input
+            type="text"
+            id="inputbox"
+            className="input-fields"
+            placeholder="test@gmail.com"
+            value={loginForm.email}
+            onChange={(e) =>
+              setLoginForm((form) => ({ ...form, email: e.target.value }))
+            }
+            required
+          />
+          <label className="password-label">Password</label>
+          <div className="visibility">
+            <input
+              type={showPassword ? "text" : "password"}
+              className="visibility-fields"
+              placeholder="test1234"
+              autoComplete="off"
+              value={loginForm.password}
```

Comment:
You can destructure more loginForm object at the top and then only utilize its properties directly.